src/main/java/org/spongepowered/api/world/ChunkPreGenerate.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.slf4j.Logger;
+import org.spongepowered.api.Game;
+import org.spongepowered.api.event.world.ChunkPreGenerationEvent;
+import org.spongepowered.api.scheduler.Scheduler;
+import org.spongepowered.api.util.ResettableBuilder;
+import org.spongepowered.api.world.storage.WorldProperties;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents an ongoing chunk pre-generation.
+ */
+public interface ChunkPreGenerate {
+
+    /**
+     * The World that this task is operating on.
+     *
+     * @return The {@link WorldProperties}
+     */
+    WorldProperties getWorld();
```

Comment:
World or WorldProperties? This should be named accordingly. Also the doc needs to be updated.